applications/TopologyOptimizationApplication/custom_elements/comparison_utilities.h

```diff
@@ -0,0 +1,342 @@
+	  {
+	    for(unsigned int j=0; j<LocalStressTensor.size2(); j++)
+	      {
+		StressTensor(i,j) = LocalStressTensor(i,j);
+	      }
+	  }
+
+	//in general coordinates:
+        double SigmaEquivalent =  (0.5)*((StressTensor(0,0)-StressTensor(1,1))*((StressTensor(0,0)-StressTensor(1,1)))+
+					 (StressTensor(1,1)-StressTensor(2,2))*((StressTensor(1,1)-StressTensor(2,2)))+
+					 (StressTensor(2,2)-StressTensor(0,0))*((StressTensor(2,2)-StressTensor(0,0)))+
+					 6*(StressTensor(0,1)*StressTensor(1,0)+StressTensor(1,2)*StressTensor(2,1)+StressTensor(2,0)*StressTensor(0,2)));
+
+	if( SigmaEquivalent < 0 )
+	  SigmaEquivalent = 0;
+
+	SigmaEquivalent = sqrt(SigmaEquivalent);
+
+	//in principal stresses:
+
+        // double tolerance  = 1e-10;
+        // double zero       = 1e-10;
+        // double NormStress = 0.00;
+
+        // CheckZeroDiagonalComponents (StressTensor);
+
+        // Vector PrincipalStress(3);
+	// noalias(PrincipalStress) = ZeroVector(3);
+
+        // NormStress =SolidMechanicsMathUtilities<double>::NormTensor(StressTensor);
+
+        // Vector MainStresses(3);
+        // noalias(MainStresses) = ZeroVector(3);	
+
+        // bool main_tensor = CheckPrincipalStresses( StressTensor );
+
+        // if(!main_tensor)
+        // {
+
+        //     if(NormStress>1e-6)
+        //     {
+        //         MainStresses = SolidMechanicsMathUtilities<double>::EigenValues(StressTensor,tolerance,zero);
+        //     }
+        //     else
+        //     {
+        //         noalias(MainStresses) = ZeroVector(3);
+        //     }
+
+        // }
+        // else
+        // {
+        //     noalias(MainStresses) = ZeroVector(3);
+        //     for(unsigned int i=0; i<StressTensor.size1(); i++)
+        //         MainStresses[i]=StressTensor(i,i);
+        // }
+
+
+        // for(unsigned int i=0; i<MainStresses.size(); i++)
+        //     PrincipalStress[i]=MainStresses[i];
+	
+
+        // double SigmaEquivalent =  (0.5)*((PrincipalStress[0]-PrincipalStress[1])*(PrincipalStress[0]-PrincipalStress[1]) +
+        //                                  (PrincipalStress[1]-PrincipalStress[2])*(PrincipalStress[1]-PrincipalStress[2]) +
+        //                                  (PrincipalStress[2]-PrincipalStress[0])*(PrincipalStress[2]-PrincipalStress[0]));
+
+
+        // SigmaEquivalent = sqrt(SigmaEquivalent);
+
+        //std::cout<<" SigmaEquivalent "<<SigmaEquivalent<<" StressVector "<<StressVector<<std::endl;
+
+        return SigmaEquivalent;
+
+        KRATOS_CATCH( "" )
+
+    };
+
+
+
+    void  CheckZeroDiagonalComponents (Matrix& StressTensor)
+    {
+        // No null diagonal terms are accepted in the eigenvalue calculation
+        for(unsigned int i=0; i<StressTensor.size1(); i++)
+        {
+            if (fabs(StressTensor(i,i))<1e-10)
+            {
+                StressTensor(i,i) = 1e-10;
+            }
+        }
+    }
+
+    bool  CheckPrincipalStresses (Matrix& StressTensor)
```

Comment:
Many of these operations alredy defined in the constitutive laws utilities @AlejandroCornejo  I don't know where are these right now